Swap again button integrated to transaction

gemdev111 opened the pull request:
Integrated swap again action button + swap header - to open swap pair scene based on transaction history
Updates, refactoring

closes #952 

images: 
<table>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/b528fa10-bb5b-485a-9acc-f5ef1f7cab50" style="width: 50%;" /></td>
    <td><img src="https://github.com/user-attachments/assets/18431784-2561-4a82-bd57-3ff2d37e845b" style="width: 50%;" /></td>
  </tr>
</table>

gemdev111 committed:
Enable swap navigation from transaction and wallet scenes

Introduces SwapNavigationStack and SwapInput to support navigation to swap flows from transaction and wallet scenes.
Updates SelectedAssetType to carry both from and to assets for swaps, and propagates isPresentingSelectedAssetType bindings to enable sheet presentation.
Refactors related view models and navigation stacks to support the new swap flow, and exposes necessary types for integration.

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Add 'Swap again' button to transaction scene

Introduces a conditional 'Swap again' button in the TransactionScene for swap transactions, based on a new showSwapAgain property in TransactionDetailViewModel. Also fixes a typo in EthereumFeeCalculator and updates navigation logic to clear and reset wallet assets after swap completion.

gemdev111 committed:
Update SwapSceneViewModel mock initialization

Refactored the mock initializer for SwapSceneViewModel to use the new input structure and preferences. This aligns the test setup with recent changes to the view model's constructor.

gemdev111 committed:
Update core

gemdev111 committed:
Merge branch 'main' into swap-again-button

gemdev111 committed:
Refactor sheet presentation state naming and logic

Renames isPresentedInfoSheet to isPresentingInfoSheet in SwapSceneViewModel and updates related usages for clarity.
Adds TODO comments in TransactionsViewModel and WalletSceneViewModel to separate sheet state logic.
Updates SwapNavigationStack to inject swapService and keystore into SwapQuoteDataProvider.
Fixes typo in EthereumFeeCalculator method name.
Localizations updates

gemdev111 committed:
Refactor asset selection and swap service injection

Simplifies asset selection logic in TransactionScene by removing an unnecessary closure. Updates
SwapNavigationStack to inject swapService directly into SwapQuotesProvider for consistency and improved dependency management.

gemdev111 committed:
Removed extra comma

DRadmir commented on Features/Transactions/Sources/Scenes/TransactionScene.swift:
```diff
@@ -100,34 +104,59 @@ public struct TransactionScene: View {
 extension TransactionScene {
-    func onNetworkFeeInfo() {
+    func onSelectTransactionHeader() {
+        switch model.headerType {
```
Lets move the logic into model and return a tuple (fromAsset, toAsset)

DRadmir commented on Gem/Navigation/Swap/SwapNavigationStack.swift:
```diff
@@ -0,0 +1,45 @@
+
+    var body: some View {
+        NavigationStack(path: $navigationPath) {
+            SwapNavigationView(
```
Should we keep both SwapNavigationView and SwapNavigationStack?

gemdev111 replied:
Sure, NavigationView it's a just view with a sheets & navigationDestination not wrapped in stack. NavigationStack different logic, only manages the navigation path

DRadmir commented on Gem/Navigation/Transactions/TransactionsNavigationStack.swift:
```diff
@@ -84,6 +86,27 @@ struct TransactionsNavigationStack: View {
+                                )
+                            ),
+                            onComplete: {
+                                navigationState.selectedTab = .wallet
```
Move to private method please

DRadmir commented on Gem/Navigation/Wallet/WalletNavigationStack.swift:
```diff
@@ -123,6 +126,25 @@ struct WalletNavigationStack: View {
+                                )
+                            ),
+                            onComplete: {
+                                navigationState.wallet.removeLast()
```
Move to private method please

gemdev111 committed:
Refactor swap asset selection and completion handling

Centralizes swap asset selection logic in TransactionDetailViewModel and refactors swap completion handlers in TransactionsNavigationStack and WalletNavigationStack into private methods for improved code clarity and reuse.

gemcoder21 reviewed (changes requested):
Need to be simplified by using deep links maybe? so you just call `/swap..` and it will open swap screen instead?

gemcoder21 commented:
Try using deeplinks in `onReceiveNotification` on `MainTabView`

gemdev111 committed:
Refactor swap navigation to use deep links and sheets

Replaces the previous swap navigation using isPresentingSelectedAssetType with a deep link and sheet-based approach.
Adds support for swap deep links in URL parsing, push notifications, and navigation.
Updates SwapInput and SwapPairSelectorViewModel for better integration and public access. Cleans up obsolete state and logic from Wallet and Transactions navigation stacks.

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Remove unused Swap import statements

Deleted unnecessary imports of the Swap module from TransactionsNavigationStack.swift and WalletNavigationStack.swift to clean up the code.

gemdev111 committed:
Refactor asset selection and swap navigation logic

Unifies asset selection sheet state management by renaming and relocating SelectedAssetInput to Primitives, and updates all references to use isPresentingSelectedAssetInput.
Removes SwapNavigationStack in favor of handling swap navigation via SelectedAssetNavigationStack.
Refactors MainTabView and related navigation to support swap actions through SelectedAssetInput
Improves URLParser error handling and test coverage for swap URLs.

gemdev111 committed:
Update core

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Refactor NFT navigation and asset selection handling

Removed CollectibleNavigationView and CollectibleNavigationViewModel, integrating asset selection and navigation logic directly into CollectibleViewModel and CollectibleScene. Updated related view models and navigation stacks to use Binding<SelectedAssetInput?> for asset selection.
Adjusted MainTabView, CollectionsNavigationStack, and SelectedAssetNavigationStack to support the new navigation flow and improved state management.

gemdev111 committed:
Removed comma, ci fix
